qt_ui/windows/AirWingConfigurationDialog.py

```diff
@@ -79,25 +84,38 @@ class SquadronBaseSelector(QComboBox):
     def __init__(
         self,
         bases: Iterable[ControlPoint],
-        squadron: Squadron,
+        selected_base: Optional[ControlPoint],
+        aircraft_type: Optional[AircraftType],
     ) -> None:
         super().__init__()
-        self.bases = list(bases)
-        self.squadron = squadron
         self.setSizeAdjustPolicy(self.AdjustToContents)
-
-        for base in self.bases:
-            if not base.can_operate(self.squadron.aircraft):
-                continue
-            self.addItem(base.name, base)
-        self.model().sort(0)
-        self.setCurrentText(self.squadron.location.name)
+        self.bases = list(bases)
+        self.set_aircraft_type(aircraft_type)
+
+        if selected_base:
+            self.setCurrentText(selected_base.name)
+        # TODO can we get a prefered base if none is selected?
```

Comment:
Yeah i think the effort to solve the "problem" is not worth it because as you said: the most people will change the base either way. I was just unsure if we already have such a function but as we do not i will just leave it as it is :)